src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -243,7 +243,7 @@ export const Map: React.FC<MapProps> = ({
         <img
           width="100%"
           className="party-map-background"
-          src={venue.mapBackgroundImageUrl ?? DEFAULT_MAP_BACKGROUND}
+          src={venue.mapBackgroundImageUrl}
```

Comment:
Since DEFAULT_MAP_BACKGROUND was removed here, I would have expected there to also be no need to import it anymore, but it doesn't appear that is the case? Where else is it used in this file, and for what purpose? It seems like it should probably be removed from everywhere..?

Reply:
The default map dimensions are being used to calculate the grid when the map background is missing. I'm not sure if this is the accurate flow, but that's why this is still being used.

Reply:
Without looking too deeply into it, I'd say that's probably not a desirable flow.

Looking at the `useEffect` code where it's used, it seems to only be used for calculating the aspect ratio of the map image (which is then used to calculate the number of rows to use). I feel like we can just as easily set that to a default (I think we even have default rows already maybe?) I expect it will be wrong and need to re-run currently anyway, so no need to do keep it around for that reason.

https://github.com/sparkletown/sparkle/blob/7d5773daa6eadd6a1faf46bafb9b521365d36a84/src/components/templates/PartyMap/components/Map/Map.tsx#L73-L87

Since we load `mapBackgroundImageUrl` into an `<img>` tag later as well (from an entirely different piece of code), it's possible that this will actually result in us attempting to fetch the map background image twice (assuming caching/etc doesn't intervene and 'save us')

I haven't tested this, but I believe we can probably handle this same 'requirement' using a ref (or a callback ref perhaps?) on the `<img>` tag. I would be looking to follow the same sort of pattern as I used in the `Audience` component to call my 'focus on first load' code.

https://github.com/sparkletown/sparkle/blob/707ab63e1bbdc2f592213be7b71a62c378a66541/src/components/templates/Audience/Audience.tsx#L160-L174

https://github.com/sparkletown/sparkle/blob/707ab63e1bbdc2f592213be7b71a62c378a66541/src/components/templates/Audience/Audience.tsx#L414-L418

Reply:
Actually.. even simpler than that, since we are using an `<img>` we can just call it from `onLoad`:

- https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload
  - > The `onload` property of the `GlobalEventHandlers` mixin is an `EventHandler` that processes load events on a `Window`, `XMLHttpRequest`, **`<img>` element**, etc.
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img
- https://www.javascriptstuff.com/detect-image-load/

Reply:
The only reason this logic is in `useEffect` instead of `onLoad` is because it needs to be retriggered if someone changes the number of grid rows in the admin. The `useEffect` will trigger again and recalculate the columns because of the dep array, but the `onLoad` runs only once, when rendered. I searched for a way to 'retrigger' `onLoad` but I couldn't find anything.

Reply:
That being said, the constant `DEFAULT_MAP_BACKGROUND` can be removed because it's not needed for the calculation.

Reply:
In my future answers I will try to be more descriptive on how and why things are written in the way they are. It will probably save you some time writing these suggestions. I understand that sometimes it's hard to understand the whole thing just from 10 lines of code.

Reply:
Makes sense.

---

> In my future answers I will try to be more descriptive on how and why things are written in the way they are. It will probably save you some time writing these suggestions. I understand that sometimes it's hard to understand the whole thing just from 10 lines of code.

Please do! This would be immensely helpful.

I've been asking people to add proper descriptions to their PRs for a while (not just linking it to a backlog issue, but actually explaining the what/why/rationale of the changes in the PR where it makes sense to), as it helps give the reviewer the required context to assess it without needing to 'brute force' or 'figure it all out from scratch'